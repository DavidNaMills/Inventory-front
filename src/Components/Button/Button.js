import React from 'react';
import classes from './button.module.scss';

const allTypes = {
    base: {
        base: classes.button__base,
        shadow: classes.button__base_shadow,
        border: classes.button__base_border,
        skeleton: classes.button__base_skeleton,
    },

    submit: {
        base: classes.button__submit,
        shadow: classes.button__submit_shadow,
        border: classes.button__submit_border,
        skeleton: classes.button__submit_skeleton,
    },

    clear: {
        base: classes.button__clear,
        shadow: classes.button__clear_shadow,
        border: classes.button__clear_border,
        skeleton: classes.button__clear_skeleton,
    },

    danger: {
        base: classes.button__danger,
        shadow: classes.button__danger_shadow,
        border: classes.button__danger_border,
        skeleton: classes.button__danger_skeleton,
    },

    information: {
        base: classes.button__information,
        shadow: classes.button__information_shadow,
        border: classes.button__information_border,
        skeleton: classes.button__information_skeleton,
    },
};


const Button = ({ config = {}, onClick, label = null, skeleton = false, raised = false, full = false, type = 'base', border = false, bold = false, wide = false, isDisabled = false, Icon = null, noStyle=null }) => {
    const fullStyle = [classes.button];
    let which = type;
    const style = {};

    if (isDisabled) {
        fullStyle.push(classes.button__disabled);
    } else if(noStyle) { fullStyle.push(classes.button_noStyle);
    } else {
        if (allTypes[type]) {
            which = type;
            fullStyle.push(allTypes[type].base);
        } else {
            which = 'base';
            fullStyle.push(allTypes['base'].base);
        }

        if (skeleton) { fullStyle.push(allTypes[which].skeleton); }
        if (raised) { fullStyle.push(allTypes[which].shadow); }
        if (border) { fullStyle.push(allTypes[which].border); }
    }
    if (full) { fullStyle.push(classes.button_full); }
    if (bold) { fullStyle.push(classes.button_bold); }
    
    
    if (wide) {
        style.width = '30%';
    }

    return (
        <button
            {...config}
            onClick={onClick}
            style={style}
            className={fullStyle.join(' ')}
            disabled={isDisabled}
        ><p>{label}</p>{Icon && <Icon />}</button>
    )
}

export default React.memo(Button);