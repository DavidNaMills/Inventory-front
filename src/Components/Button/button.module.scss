@import '../../styles/mixins/buttonMixins.module.scss';
@import '../../styles/variables/colors.module.scss';

.button{
    display: flex;
    align-content: center;
    justify-content: center;
    font-size: 16px;
    border: none;
    padding: 10px;
    padding-left: 20px;
    padding-right: 20px;
    border-radius: 2px;

    p{
        padding-top: 5px;
        padding-right: 10px;
    }

    &__disabled{
        background-color: $disabled-back;
        color: $disabled-color;
    }

    &__base{
        @include buttonStyle($base-back, $base-color);
        &:hover{
            @include onHover($base-shadow, $light-text, $base-back);
        }

        &_shadow{@include buttonShadow($base-shadow);}
        &_border{@include buildBorder($base-color);}
        &_skeleton{@include buildSkeleton($base-back, $base-color); }
    };
    
    &__submit{
        @include buttonStyle($submit-back, $submit-color);
        &:hover{
            @include onHover($submit-shadow, $light-text, $submit-back);
        }

        &_shadow{ @include buttonShadow($submit-shadow); }
        &_border{ @include buildBorder($submit-color); }
        &_skeleton{@include buildSkeleton($submit-back, $submit-color); }
    }
    
    &__clear{
        @include buttonStyle($clear-back, $clear-color);
        &:hover{
            @include onHover($clear-shadow, $clear-color, $clear-back);
        }

        &_shadow{ @include buttonShadow($clear-shadow); }
        &_border{ @include buildBorder($clear-color); }
        &_skeleton{@include buildSkeleton($clear-back, $clear-color); }
    }
    
    &__danger{
        @include buttonStyle($danger-back, $danger-color);
        &:hover{
            @include onHover($danger-shadow, $light-text, $danger-back);
        }
        &_shadow{ @include buttonShadow($danger-shadow); }
        &_border{ @include buildBorder($danger-color); }
        &_skeleton{@include buildSkeleton($danger-back, $danger-color); }
    }
    
    &__information{
        @include buttonStyle($information-back, $information-color);
        &:hover{
            @include onHover($information-shadow, $light-text, $information-back);
        }
        &_shadow{ @include buttonShadow($information-shadow); }
        &_border{ @include buildBorder($information-color); }
        &_skeleton{@include buildSkeleton($information-back, $information-color); }
    }

    &_flat{
        border: none;
        box-shadow: none;
        -webkit-box-shadow: none;
        -moz-box-shadow: none;
    }

    &_full{
        width: 100%;
    }

    &_bold{
        font-weight: bold;
    }

    &_noStyle{
        background-color: transparent;
        border: none;
        box-shadow: none;

        &:hover{
            background-color: rgb(212, 212, 212);
            padding-top: 10px;
            padding-bottom: 10px;
            padding-left: 5px;
            padding-right: 10px;
            border-radius: 100px;
        }
    }
}