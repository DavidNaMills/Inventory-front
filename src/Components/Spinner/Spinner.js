import React from 'react';
import "react-loader-spinner/dist/loader/css/react-spinner-loader.css"
import Loader from 'react-loader-spinner'
import {isMobileOnly} from 'react-device-detect';
import usePositionHook from '../../hooks/usePositionHook/usePositionHook';
import logo from '../../images/logo.jpg';
import logoSmall from '../../images/logo-small.jpg';

const base = {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    margin: '10px'
};

const background = (pos, backImg) => ({
    position: 'absolute',
    top: `${pos}px`,
    left: '0px',
    width: '90%',
    height: '100%',
    backgroundColor: 'rgba(250, 250, 250, 0.6)',
    zIndex: '9999',
    backgroundImage: backImg ? `url(${isMobileOnly? logoSmall : logo})`: null,
    backgroundRepeat: backImg ? 'no-repeat': null,
    backgroundPosition: backImg ? 'center': null,
    backgroundSize: backImg ? 'fixed': null,
    ...base
})


const Spinner = ({ withBackground = true, backImg=false }) => {
    const { posRef } = usePositionHook();
    return (
        <div style={withBackground ? background(posRef, backImg) : base}>
            <Loader
                type="CradleLoader"
                color="#a6f5ee"
                height={100}
                width={100}
                timeout={15000} //3 secs

            />
        </div>
    )
}

export default Spinner;