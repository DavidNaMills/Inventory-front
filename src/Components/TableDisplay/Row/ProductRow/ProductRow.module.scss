.productRow{
    display: flex;
    align-items: center;
    
    &__infoCont{
        display: flex;
        flex-direction: column;
        margin-left: 20px;

    }
    
    

    &__cellCont{
        width: 100%;
        padding-top: 20px;
        display: flex;

        p{
            margin-right: 50px;
        }

        p:last-of-type {
            margin-right: none;
        }

    }

    &__cellCont_stack{
        width: 100%;
        padding-top: 10px;
        padding-bottom: 10px;
        display: flex;
        flex-direction: column;

        p{
            margin-right: 50px;
        }

        p:last-of-type {
            margin-right: none;
        }

    }



    span{
        font-weight: bold;
        margin-right: 10px;
    }
    
}
