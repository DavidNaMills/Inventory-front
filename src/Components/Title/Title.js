import React from 'react';
import classes from './title.module.scss';


const Title = (props) => {
    const { size = null, message = '', center = false } = props;
    let element = null;

    switch (size) {
        case 'h1':
            element = <h1 className={[classes.title_h1, center ? classes.title__center : null].join(' ')} >{message}</h1>
            break;
        case 'h2':
            element = <h2 className={[classes.title_h2, center ? classes.title__center : null].join(' ')}>{message}</h2>
            break;
        case 'h3':
            element = <h3 className={[classes.title_h3, center ? classes.title__center : null].join(' ')}>{message}</h3>
            break;
        case 'h4':
            element = <h4 className={[classes.title_h4, center ? classes.title__center : null].join(' ')}>{message}</h4>
            break;
        case 'h5':
            element = <h5 className={[classes.title_h5, center ? classes.title__center : null].join(' ')}>{message}</h5>
            break;
        case 'h6':
            element = <h6 className={[classes.title_h6, center ? classes.title__center : null].join(' ')}>{message}</h6>
            break;
        default:
            element = <h2 className={[classes.title_h2, center ? classes.title__center : null].join(' ')}>{message}</h2>
            break;
    };

    return (
        <div>
            {element}
        </div>
    )
}

export default React.memo(Title);