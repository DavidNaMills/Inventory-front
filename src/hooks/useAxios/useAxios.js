import {useState} from 'react';

const defaultState = {
    isLoading: false,
    errors: null,
    data: null,
    id: null
}

const useAxios = (axios) =>{
    const [fetchState, setFetchState] = useState(defaultState);

    const makeCall = ({url, method, data, callBack=()=>{}, newId=null}) =>{
        const fullConfig = {
            url,
            method
        };
        if(data){
            fullConfig.data = data
        }
        setFetchState({...defaultState, isLoading: true, id: newId})

        return axios(fullConfig)
        .then(res=>{
            // setTimeout(()=>{
                const result = {
                    id: newId,
                    isLoading: false,
                    errors: null,
                    data: res.data
                }

                setFetchState(result)
                callBack(res.data);
                return result;
            // }, 3000);
        })
        .catch(err=>{
            setFetchState({...defaultState, id: newId});
            callBack({
                id: newId,
                error: true,
                message: 'went wrong',
                data: null
            });
        })
    }

    return {
        fetchState,
        makeCall
    }
}

export default useAxios;