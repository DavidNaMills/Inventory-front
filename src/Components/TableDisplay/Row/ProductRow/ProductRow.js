import React from 'react';
import { isMobileOnly } from 'react-device-detect';
import classes from './ProductRow.module.scss';
import ImageDisplay from '../../../ImageDisplay/ImageDisplay';
import Title from '../../../Title/Title';
import { useLocale } from '../../../../Context/LocaleContext';

const ProductRow = (props) => {
    const { config, data, withTitle=true, withImage=true, withInfo=true, stack=false } = props;
    const locale = useLocale();

    return (
        <div className={classes.productRow}>
            {withImage && <ImageDisplay url={data.url ? `/productImages/${data.url}` : null} alt={`${data.name}-${data.code}`} size='medium'/>}
            <div className={classes.productRow__infoCont}>
                {withTitle && <Title message={data[config.main[0]]} size='h4' />}
                {withInfo &&
                    <div className={stack? classes.productRow__cellCont_stack : classes.productRow__cellCont}>
                        <p><span>{locale[config.main[1]]}:</span>{data[config.main[1]]}</p>
                        <p><span>{locale[config.main[2]]}:</span>{data[config.main[2]]}</p>
                    </div>
                }
            </div>
        </div>
    )
}

export default ProductRow;