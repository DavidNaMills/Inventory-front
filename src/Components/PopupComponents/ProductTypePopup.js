import React, { useEffect } from 'react';
import Title from '../Title/Title';
import { isMobileOnly } from 'react-device-detect';
import classes from './Popup.module.scss';
import ProductRow from '../TableDisplay/Row/ProductRow/ProductRow';

import useAxios from '../../hooks/useAxios/useAxios';
import axios from '../../Axios/axiosInstance';
import { URLS } from '../../Axios/urls';

import Button from '../Button/Button';
import Spinner from '../../Components/Spinner/Spinner';
import {useLocale} from '../../Context/LocaleContext';

const scrollHeight = { maxHeight: '50vh' };
const config = {
    main: ['', 'name', 'qtyInStock']
}

const ProductTypePopup = (props) => {
    const { item, onClose } = props;
    const { fetchState, makeCall } = useAxios(axios);
    const locale = useLocale();

    useEffect(() => {
        makeCall({
            url: `${URLS.allProductsWithTypes}/${item._id}`,
            method: 'GET'
        });
    }, [])

    return (
        <div className={classes.popup}>
            <Title message={item.displayValue} size='h2' center />
            {
                item.description &&
                <p className={classes.popup__mediumSpacing}>{item.description}</p>
            }

            {fetchState.isLoading
                ? <Spinner withBackground={false}/>
                : <div className={classes.popup__color1}>
                    <div className={classes.popup__mediumSpacing}>
                        <Title message={`${locale['products_msg']}: ${fetchState.data ? fetchState.data.count : 0} ${locale['in_ttl']}`} size='h4' />
                    </div>
                    <div className={classes.popup__scrollable} style={scrollHeight}>
                        {
                            (fetchState.data && fetchState.data.products)
                                ? fetchState.data.products.map((c, d) =>
                                    <div key={d} className={d % 2 ? classes.popup__row_odd : classes.popup__row_even}>
                                        <ProductRow
                                            data={c}
                                            config={config}
                                            withTitle={false}
                                            withImage={isMobileOnly ? false : true}
                                            stack={isMobileOnly ? true : false}
                                        />
                                    </div>
                                )

                            : <p className={classes.popup__mediumSpacing}>{locale['no_products_type']}</p>
                        }
                    </div>
                </div>
            }
            <Button onClick={onClose} label={locale['close']} full />
        </div>
    )
}

export default ProductTypePopup;