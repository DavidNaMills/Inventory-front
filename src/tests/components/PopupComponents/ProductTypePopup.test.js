import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import * as device from 'react-device-detect';

import '../../testHelpers/context';

jest.mock('../../../hooks/useAxios/useAxios')
import useAxios from '../../../hooks/useAxios/useAxios';

import ProductTypePopup from '../../../Components/PopupComponents/ProductTypePopup';
import ProductRow from '../../../Components/TableDisplay/Row/ProductRow/ProductRow';
import Button from '../../../Components/Button/Button';
import Title from '../../../Components/Title/Title';
import Spinner from '../../../Components/Spinner/Spinner';

const item = {
    _id: '123123123',
    displayValue: 'dsadsdas'
    // description: 'ddsdsdsdsdsds'
};

const mockClose = jest.fn();
const mockCall = jest.fn();

const mockFetchState = {
    isLoading: false,
    errors: null,
    data: {
        count: 2,
        products: [
            { _id: '2121', name: 'cups', qtyInstock: '123' },
            { _id: '7894512', name: 'plates', qtyInstock: '789' },
        ]
    },
    id: null
}


describe('<ProductTypePopup /> component test suite', () => {
    describe('happy route', () => {
        let wrapper;
        let orig;
        let div;

        beforeEach(() => {

            useAxios.mockImplementation(() => ({
                fetchState: mockFetchState,
                makeCall: mockCall
            }));

            orig = device.isMobileOnly;
            wrapper = shallow(<ProductTypePopup item={item} onClose={mockClose} />);
            div = wrapper.find('div');
        });

        afterEach(() => {
            device.isMobileOnly = orig;
            jest.clearAllMocks();
        })


        it('matches snapshot', () => {
            const comp = renderer.create(<ProductTypePopup item={item} onClose={mockClose} />);
            const app = comp.toJSON();
            expect(app).toMatchSnapshot();
        });

        it('exists', () => {
            expect(wrapper.exists).toBeTruthy();
        });

        it.skip('invokes makeCall() in useAxios hook ', ()=>{
            // expect(mockCall).toHaveBeenCalledTimes(1);
            // expect(mockCall).toHaveBeenCalledTimes(1);
        });

        it('renders a <div> with classname "popup"', () => {
            expect(div.length).toBe(6);
            expect(div.at(0).props()).toHaveProperty('className', 'popup');
        });

        it('renders a <Title/> component', () => { 
            const ttl = wrapper.find(Title);
            expect(ttl.length).toBe(2);
            expect(ttl.at(0).props()).toHaveProperty('message', item.displayValue);
            expect(ttl.at(0).props()).toHaveProperty('size', 'h2');
            expect(ttl.at(0).props()).toHaveProperty('center', true);
        });

        it('does not render the description IF description is null', () => { 
            expect(wrapper.find('p').length).toBe(0);
        });

        it('renders a <div> with className "popup__color1"', () => { 
            expect(div.at(1).props()).toHaveProperty('className', 'popup__color1');
        });

        it('renders a <div> with className "popup__mediumSpacing" and <Title/> component', () => { 
            const ttl = wrapper.find(Title);
            expect(div.at(2).props()).toHaveProperty('className', 'popup__mediumSpacing');
            expect(ttl.at(1).props()).toHaveProperty('message', 'Products: 2 in total');
            expect(ttl.at(1).props()).toHaveProperty('size', 'h4');
        });
        
        it('renders a <div> with className "popup__scrollable"', () => { 
            expect(div.at(3).props()).toHaveProperty('className', 'popup__scrollable');
            expect(div.at(3).props().style).toHaveProperty('maxHeight', '50vh');

        });
        
        it('renders 2 <div>s with classnames "popup__row_odd" and "popup__row_even" and 2 <ProductRow/> components', () => { 
            expect(div.at(4).props()).toHaveProperty('className', 'popup__row_even');
            expect(div.at(5).props()).toHaveProperty('className', 'popup__row_odd');
            
            expect(div.at(4).find(ProductRow).length).toBe(1);
            expect(div.at(4).find(ProductRow).props()).toHaveProperty('withTitle', false);
            expect(div.at(4).find(ProductRow).props()).toHaveProperty('withImage', true);
            expect(div.at(4).find(ProductRow).props()).toHaveProperty('stack', false);
            
            expect(div.at(5).find(ProductRow).length).toBe(1);
            expect(div.at(5).find(ProductRow).props()).toHaveProperty('withTitle', false);
            expect(div.at(5).find(ProductRow).props()).toHaveProperty('withImage', true);
            expect(div.at(5).find(ProductRow).props()).toHaveProperty('stack', false);
        });
        
        it('renders a <Button/> component', () => { 
            const btn = wrapper.find(Button);
            expect(btn.length).toBe(1);
            expect(btn.props()).toHaveProperty('onClick', mockClose);
            expect(btn.props()).toHaveProperty('label', 'Close');
            expect(btn.props()).toHaveProperty('full', true);
        });
    });
    
    
    
    describe('variations', ()=>{
        it('renders a <div> with className "popup__mediumSpacing" and <Title/> component with message indicating 0 products', () => { 
            useAxios.mockImplementation(() => ({
                fetchState: {
                    ...mockFetchState,
                    data: {
                        count: 0,
                        products: []
                    }
                },
                makeCall: mockCall
            }));
            const wrapper = shallow(<ProductTypePopup item={item} onClose={mockClose} />)
            const ttl = wrapper.find(Title);
            expect(wrapper.find('div').at(2).props()).toHaveProperty('className', 'popup__mediumSpacing');
            expect(ttl.at(1).props()).toHaveProperty('message', 'Products: 0 in total');
            expect(ttl.at(1).props()).toHaveProperty('size', 'h4');
        });
        

        it('renders a <p> element with the description IF description is not null', () => { 
            const itemDescription = {
                _id: '123123123',
                displayValue: 'dsadsdas',
                description: 'ddsdsdsdsdsds'
            };

            useAxios.mockImplementation(() => ({
                fetchState: {
                    ...mockFetchState,
                    data: {
                        count: 0,
                        products: []
                    }
                },
                makeCall: mockCall
            }));
            const wrapper = shallow(<ProductTypePopup item={itemDescription} onClose={mockClose} />)
            expect(wrapper.find('p').length).toBe(1);
        });
        
        it('renders a <p> and a message if data property returned is null ', () => { 
            useAxios.mockImplementation(() => ({
                fetchState: {
                    ...mockFetchState,
                    data: null
                },
                makeCall: mockCall
            }));
            const wrapper = shallow(<ProductTypePopup item={item} onClose={mockClose} />)
            const p = wrapper.find('p');
            expect(p.length).toBe(1);
            expect(p.text()).toEqual('No products of this type');
        });
        
        it('renders <ProductRow/> with changes to props if isMobileOnly is true', () => { 
            const orig = device.isMobileOnly;
            device.isMobileOnly = true;
            useAxios.mockImplementation(() => ({
                fetchState: mockFetchState,
                makeCall: mockCall
            }));
            const wrapper = shallow(<ProductTypePopup item={item} onClose={mockClose} />)

            const row = wrapper.find(ProductRow);
            expect(row.length).toBe(2);
            expect(row.at(0).props()).toHaveProperty('withTitle', false);
            expect(row.at(0).props()).toHaveProperty('withImage', false);
            expect(row.at(0).props()).toHaveProperty('stack', true);
            
            expect(row.at(1).props()).toHaveProperty('withTitle', false);
            expect(row.at(1).props()).toHaveProperty('withImage', false);
            expect(row.at(1).props()).toHaveProperty('stack', true);

            device.isMobileOnly = orig;
        });
        
        it('renders <Spinner/> component if loading', () => { 
            useAxios.mockImplementation(() => ({
                fetchState: {
                    ...mockFetchState,
                    isLoading: true
                },
                makeCall: mockCall
            }));
            const wrapper = shallow(<ProductTypePopup item={item} onClose={mockClose} />);
            expect(wrapper.find('div').length).toBe(1);
            expect(wrapper.find(Spinner).length).toBe(1);
        });
    });
});