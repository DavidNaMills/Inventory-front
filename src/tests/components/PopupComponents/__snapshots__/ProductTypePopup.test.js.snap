// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductTypePopup /> component test suite happy route matches snapshot 1`] = `
<div
  className="popup"
  style={
    Object {
      "maxWidth": "100%",
    }
  }
>
  <div>
    <h2
      className="title_h2 title__center"
    >
      dsadsdas
    </h2>
  </div>
  <div
    className="popup__color1"
  >
    <div
      className="popup__mediumSpacing"
    >
      <div>
        <h4
          className="title_h4 "
        >
          Products: 2 in total
        </h4>
      </div>
    </div>
    <div
      className="popup__scrollable"
      style={
        Object {
          "maxHeight": "50vh",
        }
      }
    >
      <div
        className="popup__row_even"
      >
        <div
          className="productRow"
        >
          <img
            alt="cups-undefined"
            className="imageDisplay imageDisplay__medium"
            src="noImage.jpg"
          />
          <div
            className="productRow__infoCont"
          >
            <div
              className="productRow__cellCont"
            >
              <p>
                <span>
                  name
                  :
                </span>
                cups
              </p>
              <p>
                <span>
                  Qty
                  :
                </span>
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        className="popup__row_odd"
      >
        <div
          className="productRow"
        >
          <img
            alt="plates-undefined"
            className="imageDisplay imageDisplay__medium"
            src="noImage.jpg"
          />
          <div
            className="productRow__infoCont"
          >
            <div
              className="productRow__cellCont"
            >
              <p>
                <span>
                  name
                  :
                </span>
                plates
              </p>
              <p>
                <span>
                  Qty
                  :
                </span>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <button
    className="button button__base button_full"
    disabled={false}
    onClick={[MockFunction]}
    style={Object {}}
  >
    <p>
      Close
    </p>
  </button>
</div>
`;
